web_monitoring/cli.py

```diff
@@ -41,14 +47,72 @@ def import_ia(url, *, from_date=None, to_date=None, maintainers=None,
     _add_and_monitor(versions)
 
 
-def import_pf_archive(cabinet_id, archive_id, *, maintainers=None, tags=None):
-    # Pulling on this generator does the work.
-    versions = pf.archive_to_versions(cabinet_id, archive_id,
-                                      maintainers=maintainers,
-                                      tags=tags)
+def import_ia_db_urls(*, from_date=None, to_date=None, maintainers=None,
+                      tags=None):
+    client = db.Client.from_env()
+    domains, url_keys = _get_db_page_url_info(client)
+    print('Importing {} URLs from {} Domains:\n  {}'.format(
+        len(url_keys),
+        len(domains),
+        '\n  '.join(domains)))
+
+    versions = (ia.timestamped_uri_to_version(version.date, version.raw_url,
+                                              url=version.url,
+                                              maintainers=maintainers,
+                                              tags=tags,
+                                              view_url=version.view_url)
+                for version in _list_ia_versions_for_domains(domains,
+                                                             from_date,
+                                                             to_date,
+                                                             url_keys))
     _add_and_monitor(versions)
 
 
+def _list_ia_versions_for_domains(domains, from_date, to_date,
+                                  filter_keys=None):
+    skipped = 0
+
+    for domain in domains:
+        ia_versions = ia.list_versions(f'http://{domain}/*',
+                                       from_date=from_date,
+                                       to_date=to_date)
+        for version in ia_versions:
+            # TODO: this is an imperfect shortcut -- if the key algorithm ever
+            # changes on our side or IA's side (which seems like a given that
+            # it will *sometime*), it won't work right. Maybe the DB needs
+            # `add=existing_pages` or something?
+            if version.key in filter_keys:
+                yield version
+            else:
+                skipped += 1
+                logger.debug('Skipping URL "%s"', version.url)
+
+    if skipped > 0:
+        logger.info('Skipped %s URLs that were unknown', skipped)
+
+
+def _get_db_page_url_info(client):
+    url_keys = set()
+    domains = set()
+    for page in _list_all_db_pages(client):
+        url_keys.add(page['url_key'])
+        domains.add(HOST_EXPRESSION.match(page['url']).group(1))
+
+    return domains, url_keys
+
+
+# TODO: this should probably be a method on db.Client, but db.Client could also
+# do well to transform the `links` into callables, e.g:
+#     more_pages = pages['links']['next']()
```

Comment:
I was wary of actually doing this here because it:

- Probably has a lot of room for discussion over different APIs/implementations
- Ought to be applied across all `list_*` methods, which is not small
- Probably entails big changes because we ought to break apart the URL composition/request from the response parsing bits.

Reply:
Agreed to all of the above.